File: Data Structures & Algorithms/anagram-groups/submission-1.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<long long,int> box;
        vector<vector<string>> output;
        int k = 0;
        long long temp = 1;

        for(int i = 0;i<strs.size();i++){
            for(int j = 0;j<strs[i].length();j++){
                temp = int(strs[i][j])*temp;
            }
            
            if(!(box.count(temp))){
                box[temp] = k;
                output.push_back({strs[i]});
                k = k + 1;
            }else{
                output[box[temp]].push_back(strs[i]);
            }
            temp = 1;
        }
        return output;
    }
};

File: Data Structures & Algorithms/valid-palindrome-ii/submission-1.java
class Solution {
    public boolean validPalindrome(String s) {
        int l=0;
        int r = s.length()-1;

        int count=0;
        while(l<=r){
            if(s.charAt(l)!=s.charAt(r)){
                return helper(l,r-1,s)||helper(l+1,r,s);
            }
            l++;
            r--;
        } 
        return true;
    }

    public boolean helper(int l,int r,String s){
        while(l<=r){
            if(s.charAt(l)!=s.charAt(r)){
                return false;
            }
            l++;
            r--;
        }
        return true;
    }
}